src/gps.cpp

```diff
@@ -31,14 +31,63 @@ const String Gps::INF_SEVERITY_STRING[] = {
   String("TST"), String("DBG")
 };
 
+bool Gps::is_neo6() const {
+  if (String(hwString).substring(0,4) == String("0004")) {
+    return true;
+  }
+  return false;
+}
+
+bool Gps::is_neo8() const {
+  if (String(hwString).substring(0,4) == String("0008")) {
+    return true;
+  }
+  return false;
+}
+
+bool Gps::is_neo10() const {
+  if (String(hwString).substring(0,4) == String("000A")) {
+    return true;
+  }
+  return false;
+}
+
+String Gps::hw() const {
+  if (is_neo6()){
+    return "Neo6";
+  }
+  if (is_neo8()){
+    return "Neo8";
+  }
+  if (is_neo10()) {
+    return "NeoA";
+  }
+  return "Neo" + String(hwString).substring(3,4);
+}
+
 void Gps::begin() {
   setBaud();
   softResetGps();
   if (mGpsNeedsConfigUpdate) {
     configureGpsModule();
   }
-  enableAlpIfDataIsAvailable();
   pollStatistics();
+  if((!is_neo6()) || (!SD.exists(AID_INI_DATA_FILE_NAME))) {
+    // we're on a non-6 neo and avoid AID_INI because is deprecated
+    // or we're on a neo6 but last boot we didn't get far enough to receive fresh
+    // ALP_INI data after initializing
+    // so restart GPS for good measure.
+    if (is_neo6()) log_i("We found no AID_INI on with neo6 on boot - coldstart gps in case  its in a state where it doesn't get fixes");
+    if (!is_neo6()) log_i("Coldstart because we found that newer neos profit from that.");
+
+    coldStartGps();
+  }  
+  pollStatistics();
+
+  if (is_neo6()) {
+    enableAlpIfDataIsAvailable();
```

Comment:
@amandel this is where we only do ALP for neo6.

Reply:
We still download ALP data in server mode. We should know which version of GPS module we have at that point in time and so can skip the alp download altogether.

Reply:
I made ALP download conditional below.